File: engine/b/src/main/java/net/crazyinvestor/engine_b/entity/TickerHistory.java
package net.crazyinvestor.engine_b.entity;

import lombok.*;

import javax.persistence.*;
import java.sql.Timestamp;
import java.time.LocalDateTime;

@Entity
@Setter
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TickerHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "ticker_id")
    private Ticker ticker;

    @Column(name = "timestamp")
    private Long timestamp;

    @Column(name = "open_price")
    private Double openPrice;

    @Column(name = "high_price")
    private Double highPrice;

    @Column(name = "low_price")
    private Double lowPrice;

    @Column(name = "close_price")
    private Double closePrice;

    @Column(name = "volume")
    private Double volume;

    @Column(name = "created_at")
    private Long createdAt;

    @Column(name = "updated_at")
    private Long updatedAt;
}

File: engine/a/ticker_producer_engine/src/main/kotlin/net/crazyinvestor/ticker_producer_engine/entity/TickerHistory.kt
package net.crazyinvestor.ticker_producer_engine.entity

data class TickerHistory (
    val id: Long?,
    val ticker: Ticker?,
    val timestamp: Long?,
    val openPrice: Double?,
    val highPrice: Double?,
    val lowPrice: Double?,
    val closePrice: Double?,
    val volume: Double? ,
    val createdAt: Long?,
    val updatedAt: Long?,
)